pkg/cmd/cli/cmd/edit.go

```diff
@@ -0,0 +1,333 @@
+pass -o json.
+
+In the event an error occurs while updating, a temporary file will be created on disk
+that contains your unapplied changes. The most common error when updating a resource
+is another editor changing the resource on the server. When this occurs, you will have
+to apply your changes to the newer version of the resource, or update your temporary
+saved copy to include the latest resource version.
+
+Examples:
+
+	# Edit the service named 'docker-registry':
+	$ %[1]s edit svc/docker-registry
+
+	# Edit the deployment config named 'my-deployment':
+	$ %[1]s edit dc/my-deployment
+
+	# Use an alternative editor
+	$ OSC_EDITOR="nano" %[1]s edit dc/my-deployment
+
+	# Edit the service 'docker-registry' in JSON using the v1beta3 API format:
+	$ %[1]s edit svc/docker-registry --output-version=v1beta3 -o json
+`
+)
+
+func NewCmdEdit(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	var filenames util.StringList
+	cmd := &cobra.Command{
+		Use:   "edit -f FILENAME",
+		Short: "Edit a resource on the server and apply the update.",
+		Long:  fmt.Sprintf(edit_long, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunEdit(fullName, f, out, cmd, args, filenames)
+			if err == errExit {
+				os.Exit(1)
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: yaml|json.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	cmd.Flags().VarP(&filenames, "filename", "f", "Filename, directory, or URL to file to use to edit the resource.")
+	return cmd
+}
+
+func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, filenames util.StringList) error {
+	var p kubectl.ResourcePrinter
+	switch format := cmdutil.GetFlagString(cmd, "output"); format {
+	case "json":
+		p = &kubectl.JSONPrinter{}
+	case "yaml":
+		p = &kubectl.YAMLPrinter{}
+	default:
+		return cmdutil.UsageError(cmd, "The flag 'output' must be one of yaml|json")
+	}
+
+	cmdNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	mapper, typer := f.Object()
+	rmap := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: f.ClientMapperForCommand(),
+	}
+
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(filenames...).
+		//SelectorParam(selector).
+		ResourceTypeOrNameArgs(true, args...).
+		Latest()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	r := b.Flatten().Do()
+	obj, err := r.Object()
+	if err != nil {
+		return err
+	}
+
+	results := editResults{}
+	for {
+		// try conversion to all the possible versions
+		// TODO: simplify by adding a ResourceBuilder mode
+		defaultVersion := clientConfig.Version
```

Comment:
The `defaultVersion` is always the same.  Move outside the loop to make that clear.